account_mtd/models/mtd_exchange_authorisation.py

```diff
@@ -107,6 +107,8 @@ def handle_exchange_user_authorisation_response(
                 response_token=response_token
             )
             record.response_from_hmrc = error_message
+            if hasattr(record, 'handle_user_authorisation_error'):
+                record.handle_user_authorisation_error(record)
```

Comment:
This seems wrong. If you're calling the method against the record then self in handle_user_authorisation_error could be used instead of the argument record.